File: engine/example/src/main/scala/pl/touk/nussknacker/engine/example/UtilProcessHelper.scala
package pl.touk.nussknacker.engine.example

import io.circe.Encoder
import pl.touk.nussknacker.engine.api.{Documentation, ParamName}

import scala.util.Random

object UtilProcessHelper {

  import scala.collection.JavaConverters._

  @Documentation(description = "Convert map to JSON")
  def mapToJson(map: java.util.Map[String, String]): String = {
    Encoder.encodeMap[String, String].apply(map.asScala.toMap).noSpaces
  }

  @Documentation(description = "Get random number")
  def random(@ParamName("To (exclusive)") to: Int): Unit = {
    Random.nextInt(to)

  }
}

File: engine/example/src/main/java/pl/touk/nussknacker/engine/javaexample/UtilProcessHelper.java
package pl.touk.nussknacker.engine.javaexample;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

public class UtilProcessHelper {

    public static String mapToJson(java.util.Map<String, String> map) throws IOException {
        return new ObjectMapper().writeValueAsString(map);
    }
}
